src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsAssemblyRewriter.cs

```diff
@@ -140,6 +140,13 @@ public void Rewrite (bool brokenExceptionTransitions)
 				CopyFile (output, path);
 				RemoveFile (output);
 
+				// Assemblies are modified in-place, which updates the modified time. We need
+				// to touch the .jlo.xml file so that we will not re-run the JLO scanner on
+				// the next incremental build.
+				// TODO: This task should probably create copies of the assemblies instead of modifying them in place.
+				var jlo_xml_file = Path.ChangeExtension (path, ".jlo.xml");
+				File.SetLastWriteTimeUtc (jlo_xml_file, DateTime.UtcNow);
```

Comment:
Should this actually happen in the place these files are written?

Above could do something like:
```csharp
if (!Files.CopyIfStreamChanged (sw.BaseStream, destinationJLOXml)) {
        // If the file was written, the lastwritetime is updated. Touch the file otherwise.
        File.SetLastWriteTimeUtc (destinationJLOXml, DateTime.UtcNow);
}
```

Reply:
The issue is that `MarshalMethodsAssemblyRewriter` is modifying the `.dll` in-place much later in the build process (`_GenerateJavaStubs`) than the `.jlo.xml` is being written `(_LinkAssembliesNoShrink)`. This causes it to always be rescanned on the next incremental build of `_LinkAssembliesNoShrink`.

There are 2 other possible fixes for this:
- If we remove the `.jlo.xml` files as Outputs for `_LinkAssembliesNoShrink` then this will not be needed.
- When `MarshalMethodsAssemblyRewriter` gets moved to `_LinkAssembliesNoShrink` it can probably be ordered so that it isn't modifying the assembly after it is being scanned for JLO.

Reply:
Removed as no longer needed.